iroh-store/src/store.rs

```diff
@@ -105,15 +108,44 @@ impl Store {
         })
         .await??;
 
-        Ok(Store {
-            inner: Arc::new(InnerStore {
-                content: db,
-                config,
-                next_id: next_id.into(),
-            }),
-        })
+        let rpc_addr = config.rpc.listen_addr;
+
+        let inner = Arc::new(Mutex::new(InnerStore {
+            content: db,
+            config,
+            next_id: next_id.into(),
+        }));
+
+        let inner_cl = Arc::clone(&inner);
+        tokio::spawn(async move {
+            // TODO: handle error
+            rpc::new(rpc_addr, inner_cl).await.unwrap()
+        });
+        Ok(Store { inner })
```

Comment:
Is this the correct orientation of how to share the resources between the main store & the store we need access to in the rpc server? Or should there only be one way to access the store: through an rpc connection?

TBH I'm not totally sure how this is meant to be used in production.

Reply:
I would move the rpc server parts into main.rs as this is only used when run as a standalone binary